projects/kit/src/lib/utils/date/tests/append-date.spec.ts

```diff
@@ -0,0 +1,181 @@
+import {describe, expect, it} from '@jest/globals';
+
+import {appendDate} from '../append-date';
+
+const y2000m6d15 = new Date(2000, 6, 15);
+
+describe('appendDate', () => {
+    it('year: 2000, month: 6, day: 15, if {} was passed', () => {
+        const result = appendDate(y2000m6d15, {});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(15);
+    });
+
+    it('year: 2000, month: 6, day: 15, if {year: 0} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: 0});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(15);
+    });
+
+    it('year: 2000, month: 6, day: 15, if {year: 0, month: 0} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: 0, month: 0});
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(15);
+    });
+
+    it('year: 2000, month: 6, day: 15, if {year: 0, month: 0, day: 0} was passed', () => {
+        const result = appendDate(y2000m6d15, {
+            year: 0,
+            month: 0,
+            day: 0,
+        });
+
+        expect(result.getFullYear()).toBe(2000);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(15);
+    });
+
+    it('year: 2005, month: 6, day: 14, if {year: 5} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: 5});
+
+        expect(result.getFullYear()).toBe(2005);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(14);
+    });
+
+    it('year: 1995, month: 6, day: 14, if {year: -5} was passed', () => {
+        const result = appendDate(y2000m6d15, {year: -5});
+
+        expect(result.getFullYear()).toBe(1995);
+        expect(result.getMonth()).toBe(6);
+        expect(result.getDate()).toBe(14);
```

Comment:
Should it be 16 ?

```suggestion
    it('year: 2005, month: 6, day: 14, if {year: 5} was passed', () => {
        const result = appendDate(y2000m6d15, {year: 5});

        expect(result.getFullYear()).toBe(2005);
        expect(result.getMonth()).toBe(6);
        expect(result.getDate()).toBe(14);
    });

    it('year: 1995, month: 6, day: 14, if {year: -5} was passed', () => {
        const result = appendDate(y2000m6d15, {year: -5});

        expect(result.getFullYear()).toBe(1995);
        expect(result.getMonth()).toBe(6);
        expect(result.getDate()).toBe(16);
```

Taiga UI behaves in the such way